core/src/main/java/org/infinispan/distribution/ch/impl/CRC16HashFunctionPartitioner.java

```diff
@@ -11,6 +11,41 @@
  */
 public class CRC16HashFunctionPartitioner extends HashFunctionPartitioner {
 
+   private boolean isPow2;
+
+   public static CRC16HashFunctionPartitioner instance(int numSegments) {
+      CRC16HashFunctionPartitioner partitioner = new CRC16HashFunctionPartitioner();
+      partitioner.init(numSegments);
+      return partitioner;
+   }
+
+   @Override
+   protected void init(int numSegments) {
+      super.init(numSegments);
+      isPow2 = isPow2(numSegments);
+   }
+
+   private static boolean isPow2(int n) {
```

Comment:
there's a nice trick for this: https://stackoverflow.com/a/600306
since it's for ulong an additional check n>0 is needed here

Reply:
Number of segments cannot ever be 0 or less, so we shouldn't even need the greater than check either.

Reply:
Nice.